libscidavis/src/Fit.cpp

```diff
@@ -603,14 +602,8 @@ void Fit::generateFitCurve(const vector<double> &par)
     if (d_gen_function) {
         insertFitFunctionCurve(objectName() + tr("Fit"), X, Y);
         d_graph->replot();
-    } else {
-        auto x = new double[d_points];
-        std::copy(X.cbegin(), X.cend(), x);
-        auto y = new double[d_points];
-        std::copy(Y.cbegin(), Y.cend(), y);
-        // according to addFitCurve docstring "..and frees the input data from memory."
-        d_graph->addFitCurve(addResultCurve(x, y));
-    }
+    } else
+        d_graph->addFitCurve(addResultCurve(X, Y));
```

Comment:
Another interesting story. There was a comment "according to addFitCurve docstring ..and frees the input data from memory." But I'm almost sure that it is does not. `addResultCurve` has two overloads, and both of them do `memcpy`, leaving the input data intact.
`addFitCurve` is implemented as
`void Graph::addFitCurve(QwtPlotCurve *c)
{
    if (c)
        d_fit_curves << c;
}`
So the curve is placed into a list. I suppose that the data inside `QwtPlotCurve` will be freed upon destruction, but it is not the same data (memory-wise) that passed to `addResultCurve`.
I think that it was a memory leak, but again, I'd like to get another opinion here. Thanks.

Reply:
It took a bit of drilling into the code, including looking up the Qwt5 setData call, but I do believe you're right. Let's lock it is.